lib/internal/streams/writable.js

```diff
@@ -290,20 +291,15 @@ ObjectDefineProperties(WritableState.prototype, {
 });
 
 function WritableState(options, stream, isDuplex) {
-  // Duplex streams are both readable and writable, but share
-  // the same options object.
-  // However, some cases require setting options to different
-  // values for the readable and the writable sides of the duplex stream,
-  // e.g. options.readableObjectMode vs. options.writableObjectMode, etc.
-  if (typeof isDuplex !== 'boolean')
-    isDuplex = stream instanceof Stream.Duplex;
-
   // Bit map field to store WritableState more effciently with 1 bit per field
   // instead of a V8 slot per field.
   this[kState] = kSync | kConstructed | kEmitClose | kAutoDestroy;
 
-  if (options && options.objectMode) this[kState] |= kObjectMode;
-  if (isDuplex && options && options.writableObjectMode) this[kState] |= kObjectMode;
+  if (options && options.objectMode)
+    this[kState] |= kObjectMode;
+
+  if (isDuplex && options && options.writableObjectMode)
+    this[kState] |= kObjectMode;
```

Comment:
Can we use a modern way of optional chaining `(?.)`:
```suggestion
  if (options?.objectMode)
    this[kState] |= kObjectMode;

  if (isDuplex && options?.writableObjectMode)
    this[kState] |= kObjectMode;
```
or are there any compatibility issues?